src/VCEL.Core/Expression/Impl/UpCastEx.cs

```diff
@@ -0,0 +1,175 @@
+﻿namespace VCEL.Core.Expression.Impl
+{
+    public static class UpCastEx
+    {
+        public static bool UpCast(ref object l, ref object r)
```

Comment:
Hand coded implementation for performance.
This method takes two parameters and promotes them to the same type if it can. It returns false if it cannot.